server/sheet-data.js

```diff
@@ -5,96 +5,319 @@ const config = require('./configuration');
+    // Further Notes
+    annotation.notes || ''
+  ];
 
-    for (var i = 0; i < 5; i++) {
-        text += possible.charAt(Math.floor(Math.random() * possible.length));
+  return appendRowToSheet(
+    row,
+    config.baseConfiguration().GOOGLE_IMPORTANT_CHANGE_SHEET_ID,
+    'A6:AN6'
+  )
+    .then(() => 'Successfully appended');
+}
+
+/**
+ * Append a row to a google sheet
+ *
+ * @param {string[]} values Column values of row  to append
+ * @param {string}   spreadsheedId ID of Google sheet to append to
+ * @param {string}   [range] Range identifying the data to append to in sheet,
+ *   e.g. `A3:Z3`. This is NOT the range after which to append -- Google sheets
+ *   identifies a continuous series of rows that intersect with this range and
+ *   appends *after those rows.*
+ * @returns {Promise<Object>} Response data from Google Sheets
+ */
+function appendRowToSheet(values, spreadsheetId, range = 'A3:ZZZ') {
+  return addAuthentication({
+    spreadsheetId,
+    // supply a cell where data exists, Google decides for itself where the data table ends and appends, using extreme range again to grab everything
+    range,
+    resource: {
+      values: [values]
+    },
+    valueInputOption: 'RAW',
+    insertDataOption: 'INSERT_ROWS'
+  })
+    .then(promisable(sheets.spreadsheets.values.append));
+}
+
+
+// ------------------- PRIVATE UTILITIES -----------------------
+
+let authTokens;
+let authClient;
+
+/**
+ * Add authentication information to a request object for Google API calls.
+ * If there is an authenticated client on hand, it will be used. Otherwise,
+ * a new authenticated client will be created and logged in before returning
+ * the object with authentication information.
+ * @private
+ * @param {Object} requestData
+ * @returns {Promise<Object>} requestData modified with auth client
+ */
+function addAuthentication (requestData) {
+  return new Promise((resolve, reject) => {
+    if (!authTokens || Date.now() > authTokens.expiry_date - 30000) {
+      const configuration = config.baseConfiguration();
+      const clientEmail = configuration.GOOGLE_SERVICE_CLIENT_EMAIL;
+      // replaces \n in .env variable with actual new lines, which the auth client expects
+      const privateKey = configuration.GOOGLE_SHEETS_PRIVATE_KEY.replace(/\\n/g, '\n');
+
+      authClient = new google.auth.JWT(
+        clientEmail,
+        null,
+        privateKey,
+        ['https://www.googleapis.com/auth/spreadsheets']);
+
+      authClient.authorize((error, tokens) => {
+        if (error) return reject(error);
+        authTokens = tokens;
+        resolve(Object.assign({auth: authClient}, requestData));
+      });
+    }
+    else {
+      resolve(Object.assign({auth: authClient}, requestData));
     }
+  });
+}
+
+/**
+ * Convert a callback-based async function into a promise-based function.
+ * @param {Function} functionWithCallback
+ * @returns {Function} Function that returns a promise
+ */
+function promisable (functionWithCallback) {
```

Comment:
This `promisable` function is really cool and I'm gonna have to save and use this in other places!